test/tests/interfacekernels/1d_interface/coupled_value_coupled_flux_with_jump_material.i

```diff
@@ -0,0 +1,125 @@
+[Mesh]
+  type = GeneratedMesh
+  dim = 1
+  nx = 10
+  xmax = 2
+[]
+
+[MeshModifiers]
+  [./subdomain1]
+    type = SubdomainBoundingBox
+    bottom_left = '1.0 0 0'
+    block_id = 1
+    top_right = '2.0 1.0 0'
+  [../]
+  [./interface]
+    type = SideSetsBetweenSubdomains
+    depends_on = subdomain1
+    master_block = '0'
+    paired_block = '1'
+    new_boundary = 'master0_interface'
+  [../]
+[]
+
+[Variables]
+  [./u]
+    order = FIRST
+    family = LAGRANGE
+    block = '0'
+  [../]
+
+
+  [./v]
+    order = FIRST
+    family = LAGRANGE
+    block = '1'
+  [../]
+[]
+
+[Kernels]
+  [./diff_u]
+    type = CoeffParamDiffusion
+    variable = u
+    D = 4
+    block = 0
+  [../]
+  [./diff_v]
+    type = CoeffParamDiffusion
+    variable = v
+    D = 2
+    block = 1
+  [../]
+[]
+
+[InterfaceKernels]
+  [./penalty_interface]
+    type = PenaltyInterfaceDiffusion
+    variable = u
+    neighbor_var = v
+    boundary = master0_interface
+    penalty = 1e6
```

Comment:
It makes more sense to me to change penalty to an interface material property. `jump_prop_name` makes this kernel too toylish.

Reply:
It is a toy though; it's in the test directory 😄